iroh-relay/src/relay_map.rs

```diff
@@ -52,50 +51,81 @@ impl RelayMap {
     pub fn is_empty(&self) -> bool {
         self.nodes.is_empty()
     }
+}
 
-    /// Creates a new [`RelayMap`] with a single relay server configured.
-    ///
-    /// Allows to set a custom STUN port and different IP addresses for IPv4 and IPv6.
-    /// If IP addresses are provided, no DNS lookup will be performed.
-    ///
-    /// Sets the port to the default [`DEFAULT_RELAY_QUIC_PORT`].
-    pub fn default_from_node(url: RelayUrl, stun_port: u16) -> Self {
-        let mut nodes = BTreeMap::new();
-        nodes.insert(
-            url.clone(),
-            RelayNode {
-                url,
-                stun_only: false,
-                stun_port,
-                quic: Some(RelayQuicConfig::default()),
-            }
-            .into(),
-        );
-
-        RelayMap {
-            nodes: Arc::new(nodes),
+impl FromIterator<RelayNode> for RelayMap {
+    fn from_iter<T: IntoIterator<Item = RelayNode>>(iter: T) -> Self {
+        Self {
+            nodes: Arc::new(
+                iter.into_iter()
+                    .map(|node| (node.url.clone(), Arc::new(node)))
+                    .collect(),
+            ),
         }
     }
+}
 
-    /// Returns a [`RelayMap`] from a [`RelayUrl`].
-    ///
-    /// This will use the default STUN port, the default QUIC port
-    /// (as defined by the `iroh-relay` crate) and IP addresses
-    /// resolved from the URL's host name via DNS.
-    /// relay nodes are specified at <../../docs/relay_nodes.md>
-    pub fn from_url(url: RelayUrl) -> Self {
-        Self::default_from_node(url, DEFAULT_STUN_PORT)
-    }
-
-    /// Constructs the [`RelayMap`] from an iterator of [`RelayNode`]s.
-    pub fn from_nodes<I: Into<Arc<RelayNode>>>(value: impl IntoIterator<Item = I>) -> Result<Self> {
-        let mut map = BTreeMap::new();
-        for node in value.into_iter() {
-            let node = node.into();
-            ensure!(!map.contains_key(&node.url), "Duplicate node url");
-            map.insert(node.url.clone(), node);
+/// Create a `RelayMap` from a `RelayUrl`.
```

Comment:
Let's make these links in the docs:
```suggestion
/// Creates a [`RelayMap`] from a [`RelayUrl`].
```